sql/sys_vars.cc

```diff
@@ -7173,3 +7173,23 @@ static Sys_var_optimizer_cost Sys_optimizer_scan_cost(
   CMD_LINE(REQUIRED_ARG),
   VALID_RANGE(0, 100000000), DEFAULT(DEFAULT_TABLE_SCAN_SETUP_COST),
   COST_ADJUST(1000));
+
+static Sys_var_charptr_fscs Sys_server_redirect_target(
+	"server_redirect_target",
+        "Server redirection target. This should be a hostname, an IP address, or "
+        "a hostname or IP address followed by ':PORT'. Server redirection will "
+        "not be activated unless SERVER_REDIRECT_MODE is also set.",
```

Comment:
On [the mailing list](https://lists.mariadb.org/hyperkitty/list/developers%40lists.mariadb.org/message/QZDOCLK7TRTSAZOJFNANX5VLBY4PKS45/), @vuvova wrote:

> I'd really rather avoid introducing a new connection string format.
> …
> MDEV-15935 says that we "might" optionally implement URI validation in
the server. Without it one can literally send an arbitrary string.
This might be error-prone. With the server side URI validation
we can limit redirection URIs to "mysql://host[:port]" which basically
is what you wanted and postpones all the URI complexity till we have
some standard specs for it.

I agree.

Allowing the server to send an arbitrary string here is error-prone, and introducing Yet Another Connection String Format seems unwise, even in the name of simplicity.

So what I should do is…

1. Change this help text to explain the format as `mysql://host[:port]`, where `host` can be either a hostname or a URL-format IP address (bracket-enclosed for IPv6, e.g. `[2000::dead:beef]`), and where `port` defaults to 3306 if unspecified.
2. Add a validation function to prevent the `SERVER_REDIRECT_TARGET` sysvar from being set to a value not matching this format.
3. Modify https://github.com/mariadb-corporation/mariadb-connector-c/pull/226 to expect this format.